core/src/main/java/org/hibernate/ogm/type/impl/ByteArrayStringType.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.type.impl;
+
+import org.hibernate.MappingException;
+import org.hibernate.engine.spi.Mapping;
+import org.hibernate.ogm.type.descriptor.impl.StringMappedGridTypeDescriptor;
+import org.hibernate.type.descriptor.java.ByteArrayTypeDescriptor;
+
+/**
+ * @author Davide D'Alto
+ */
+public class ByteArrayStringType extends AbstractGenericBasicType<Byte[]> {
+
+	public static final ByteArrayStringType INSTANCE = new ByteArrayStringType();
+
+	public ByteArrayStringType() {
+		super( StringMappedGridTypeDescriptor.INSTANCE, ByteArrayTypeDescriptor.INSTANCE );
+	}
+
+	@Override
+	public String getName() {
+		return "Byte_array";
```

Comment:
Is the uppercase "Byte" expected? If so, I'm wondering if the other types should be consistent with this naming scheme. See ByteMappedType for instance.

Shouldn't we add the _as_string suffix to be consistent with PrimitiveByteArrayStringType?